src/Extension/CoreExtension.php

```diff
@@ -349,7 +349,7 @@ function twig_cycle($values, $position)
 function twig_random(Environment $env, $values = null, $max = null)
 {
     if (null === $values) {
-        return null === $max ? mt_rand() : mt_rand(0, $max);
+        return null === $max ? mt_rand() : mt_rand(0, (int) $max);
```

Comment:
The parameter is types as `int|null` so this is a float here this would be a user error and maybe we should keep the warning ?

Reply:
But then again, there is a test case that explicitly tests for the behavior of that function when passing a float. That made me think that the silent cast was somewhat expected behavior.

The function is registered for use in Twig templates and I figured it might be better to suppress the warning than to have people do type casts in templates.

But I'm not using this function and I did this change mainly to get the tests green. I might be horribly wrong with my assessment. If we revert here, how shall we proceed with the test case?

Reply:
Well maybe the fix is just to change the type of the argument to show that we allow floats here, if we have a test with a float value this means such a value is expected.